### PR TITLE
Prevent leaking of variables to global scope

### LINKED ISSUE #28110: [Issue] Prevent leaking of variables to global scope
This issue is automatically created based on existing pull request: magento/magento2#27622: Prevent leaking of variables to global scope

---------

### Description (*)
This PR adjusts minification file resolver for RequreJS that is generated to prevent leaking of variables it uses to global scope.

### Steps to reproduce (*)

1. Enable JS minification  https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/themes/js-bundling.html

### Expected Result (*)
1. JavaScript minification still works properly and correct `.min.js` files are requested.
2. There is no `ctx`, `origNameToUrl` and `baseUrl` variables under `window` leaked.

### Actual result (*)

1. [Screenshots, logs or description]
2.

### PR DESCRIPTION
**Pull request by krzksz**
### Description (*)
This PR adjusts minification file resolver for RequreJS that is generated to prevent leaking of variables it uses to global scope.

### Manual testing scenarios (*)

1. JavaScript minification still works properly and correct `.min.js` files are requested.
2. There is no `ctx`, `origNameToUrl` and `baseUrl` variables under `window` leaked.

### Resolved issues:
1. [x] resolves magento/magento2#28110: Prevent leaking of variables to global scope

### CODE REVIEW

File: lib/internal/Magento/Framework/RequireJs/Config.php [hunk 1]
```diff
@@ -293,24 +293,26 @@ protected function getConfigFileName()
      */
     public function getMinResolverCode()
     {
-        $excludes = ['url.indexOf(baseUrl) === 0'];
+        $excludes = ['url.indexOf(baseUrl)===0'];
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/RequireJs/Config.php:296]: Why did you remove spaces here?

File: lib/internal/Magento/Framework/RequireJs/Test/Unit/ConfigTest.php [hunk 2]
```diff
@@ -137,7 +137,7 @@ function ($file) {
             ->willReturn(true);
 
         $expected = <<<expected
-(function(require){
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/RequireJs/Test/Unit/ConfigTest.php]: Unit test fails, please double check it.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/RequireJs/Config.php (hunk 1 above)
Comment (krzksz) [reply lib/internal/Magento/Framework/RequireJs/Config.php:296]: This is for consistency as the other parts of the condition added by https://github.com/magento/magento2/pull/27622/files#diff-c0df1f93853550f5aa5d365ef36c307aR298 don't have spaces between them.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7286](https://jira.corp.magento.com/browse/ENGCOM-7286) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

JS Minification still works as previously(expected)

**Review by ihor-sviziev — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7286](https://jira.corp.magento.com/browse/ENGCOM-7286) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
Nothing has changed, so no need to retest

**Comment by sdzhepa** [comment]
@magento create issue

**Review by slavvka — CHANGES_REQUESTED** [review]
@krzksz please resolve the conflicts and re-run tests after upgrade to PHP 7.4

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-7286](https://jira.corp.magento.com/browse/ENGCOM-7286) has been created to process this Pull Request

**Pull request merged**